src/Http/HttpClientFactory.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+namespace Microsoft\Graph\Http;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\RequestOptions;
+use Http\Adapter\Guzzle7\Client as GuzzleAdapter;
+use Http\Promise\Promise;
+use Microsoft\Graph\Core\NationalCloud;
+use Microsoft\Graph\Exception\ClientInitialisationException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Class HttpClientFactory
+ *
+ * Configures a Guzzle HTTP client for use with Graph API
+ *
+ * @package Microsoft\Graph\Http
+ * @copyright 2021 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+final class HttpClientFactory
+{
+    /**
+     * @var int Default connection timeout
+     */
+    const CONNECTION_TIMEOUT_SEC = 30;
+
+    /**
+     * @var int Default request timeout
+     */
+    const REQUEST_TIMEOUT_SEC = 100;
+
+    /**
+     * @var string Graph API host to use as base URL and for authentication
+     */
+    private static $nationalCloud = NationalCloud::GLOBAL;
+
+    /**
+     * @var array Guzzle client request options
+     */
+    private static $clientConfig = [];
```

Comment:
Can you document a link to the Guzzle client options in the summary?

Reply:
Adding this.